### PR TITLE
Add `css` format

### PR DESCRIPTION
**Pull request by dated**
Adds the `css` format option. This PR is mostly a duplicate of #9.

```
hexRgb("#cd2222cc", { format: "css" });
//=> "rgb(205 34 34 / 80%)"
```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -42,9 +42,12 @@ hexRgb('#cd2222cc');
...
 hexRgb('#cd2222cc', {format: 'array'});
 //=> [205, 34, 34, 0.8]
+
+hexRgb('#cd2222cc', {format: 'css'});
+//=> 'rgba(205, 34, 34, 0.8)'
 ```
 */
 declare function hexRgb(hex: string): hexRgb.RgbaObject;
-declare function hexRgb(hex: string, options: hexRgb.Options): hexRgb.RgbaTuple;
+declare function hexRgb(hex: string, options: hexRgb.Options): hexRgb.RgbaTuple | string;
```
Comment (sindresorhus) [inline index.d.ts]: The return value should be specifically typed per `format` option value, so the user gets the correct return value for the specific option.

File: index.d.ts [hunk 2]
```diff
@@ -42,9 +42,12 @@ hexRgb('#cd2222cc');
 
 hexRgb('#cd2222cc', {format: 'array'});
 //=> [205, 34, 34, 0.8]
+
+hexRgb('#cd2222cc', {format: 'css'});
+//=> 'rgba(205, 34, 34, 0.8)'
```
Comment (sindresorhus) [inline index.d.ts]: I think it should use the modern CSS format: https://twitter.com/alexjmedick/status/1254153232823263235

**Comment by dated** [comment]
All done, let me know if you want me to limit the decimal places of the alpha channel!

File: index.d.ts [hunk 3]
```diff
@@ -42,9 +47,15 @@ hexRgb('#cd2222cc');
...
+
+hexRgb('#cd2222cc', {format: 'css'});
+//=> 'rgb(205 34 34 / 80%)'
 ```
 */
 declare function hexRgb(hex: string): hexRgb.RgbaObject;
+
+declare function hexRgb(hex: string, options: hexRgb.Options): hexRgb.RgbaObject;
 declare function hexRgb(hex: string, options: hexRgb.Options): hexRgb.RgbaTuple;
+declare function hexRgb(hex: string, options: hexRgb.Options): string;
```
Comment (sindresorhus) [inline index.d.ts]: This is not correct. Like I said, the return value should be overloaded based on the format option value.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 3 above)
Comment (dated) [reply index.d.ts]: I hope f7ee876 is what you are looking for, if not I'm afraid I don't know how to achieve that 🤔

File: index.test-d.ts [hunk 4]
```diff
@@ -2,4 +2,6 @@ import {expectType} from 'tsd';
 import hexRgb = require('.');
 
 expectType<hexRgb.RgbaObject>(hexRgb('#cd2222cc'));
-expectType<hexRgb.RgbaTuple>(hexRgb('#cd2222cc', {format: 'array'}));
+expectType<hexRgb.RgbaObject | hexRgb.RgbaTuple | string>(hexRgb('#cd2222cc', {format: 'array'}));
```
Comment (sindresorhus) [inline index.test-d.ts]:
```suggestion
expectType<hexRgb.RgbaTuple>(hexRgb('#cd2222cc', {format: 'array'}));
```

Same with the others.

**Comment by sindresorhus** [comment]
> let me know if you want me to limit the decimal places of the alpha channel!

Yeah, let's limit it to 2. Should be documented.

File: index.d.ts (hunk 3 above)
Comment (sindresorhus) [reply index.d.ts]: Yes, that's what I had in mind.

File: readme.md [hunk 5]
```diff
@@ -35,8 +35,13 @@ hexRgb('#cd2222cc');
 
 hexRgb('#cd2222cc', {format: 'array'});
 //=> [205, 34, 34, 0.8]
+
+hexRgb('#cd2222cc', {format: 'css'});
+//=> 'rgb(205 34 34 / 80%)'
 ```
 
+Please note when using the `css` format the value of the alpha channel is rounded to two decimal places.
```
Comment (sindresorhus) [inline readme.md]: Readme and index.d.ts should be in sync.

Comment (sindresorhus) [inline readme.md]:
```suggestion
Please note that when using the `css` format, the value of the alpha channel is rounded to two decimal places.
```

**Comment by Delapouite** [comment]
For people coming from search engines not familiar with this new notation without comma and a slash https://css-tricks.com/no-comma-color-functions-in-css/

**Comment by alexcroox** [comment]
It would be nice to have a version with commas for maximum browser support

**Pull request merged**